modules/fvad/fvad.c

```diff
@@ -0,0 +1,348 @@
+/**
+ * @file fvad.c  Voice Activity Detection
+ *
+ * Uses libfvad from https://github.com/dpirch/libfvad
+ *
+ * Copyright (C) 2023 Lars Immisch
+ */
+#include <string.h>
+#include <stdlib.h>
+#include <re.h>
+#include <rem.h>
+#include <baresip.h>
+#include <fvad.h>
+#include "fvad.h"
+
+
+/**
+ * @defgroup fvad fvad
+ *
+ * Voice Activity Detection for the audio-signal.
+ *
+ * It is using the aufilt API to get the audio samples.
+ */
+
+
+struct vad_enc {
+	struct aufilt_enc_st af;  /* inheritance */
+	bool vad_tx;
+	Fvad *fvad;
+	const struct call *call;
+};
+
+
+struct vad_dec {
+	struct aufilt_enc_st af;  /* inheritance */
+	bool vad_rx;
+	Fvad *fvad;
+	const struct call *call;
```

Comment:
`const` does not have benefits here because it has to be cast away for `uag_filter_calls()`.